### DISCUSSION TITLE
install omlib-modelica-4.0.0 in docker

### DISCUSSION THREAD

**Discussion by mwetter** [OpenModelica/OpenModelica#9357] [Q&A] [answered]
We used to install `omlib-modelica-4.0.0=4.0.0~20210622~131817~git~OM~maint~4.0.x-1` with `apt-get` in a docker: https://github.com/lbl-srg/docker-ubuntu-omc/blob/d75eec38e24b5308ffff58432d3115c2e84ea7da/Dockerfile#L30

This package (or any package `omlib-modelica-4.0.0`) can no longer be found.

How can we programmatically install MSL 4.0.0 and its dependencies in a docker file?  Pulling the repository information as described in https://openmodelica.org/index.php/download/download-linux does not resolve the problem.

**Comment by adrpo** [comment, marked as answer]
See https://www.openmodelica.org/doc/OpenModelicaUsersGuide/omlib-staging/packagemanager.html

You can now install the libraries programmatically via `omc `using a script `installPackage.mos` containing:
// --- install Modelica 4.0.0
updatePackageIndex(); getErrorString();
installPackage(Modelica, "4.0.0"); getErrorString();
// --- end install 

The libraries are no longer installed system wide but locally for the user in `~/.openmodelica`
so one might need to add a user from which you run omc or share a directory from the host 
containing the libraries.

@sjoelund are there any things I forgot?

**Reply by mwetter** [reply]
Thanks @adrpo , that works. For anyone else interested, the key lines in the docker are here:
https://github.com/lbl-srg/docker-ubuntu-omc/blob/490eb997e4bffda00adec07597ee895a83dfcaad/Dockerfile#L55
and, as we run the docker on travis which happens to have another user id than the user inside the docker, the permissions had to be set on the working directory to which omc writes output, as in
https://github.com/lbl-srg/modelica-buildings/blob/61a8b5b4c95e56a817abfdee25b4db9d5033b270/Buildings/Resources/Scripts/travis/omc/omc#L89

**Reply by sjoelund** [reply]
This reminds me that it would be convenient to have command-line flags to install packages :)
`omc --installPackages "Modelica 4.0.0" "Buildings 9"` or something

**Reply by mwetter** [reply]
Yes, that would be more convenient than the current approach.

**Reply by casella** [reply]
Se #9436
